### ISSUE TITLE
prefer-to-contain: autofix produces syntactically invalid code for optional-chained includes (arr?.includes)

### ISSUE THREAD

**Issue by Meet6338-X** [jest-community/eslint-plugin-jest#2007]
## Summary

The `prefer-to-contain` autofixer removes the `includes(...)` call by slicing from `includesCall.callee.property.range[0] - 1`, which assumes the member access is separated from the object by exactly one `.` character. For **optional chaining** (`expect(arr?.includes('x')).toBe(true)`), that slice leaves the trailing `?` behind, and the produced fix is syntactically invalid JavaScript:

```diff
- expect(arr?.includes('x')).toBe(true);
+ expect(arr?).toContain('x');   // SyntaxError
```

Running `eslint --fix` therefore corrupts the file.

*Note: this is a static analysis finding based on reading `main`; I did not execute ESLint for this report.*

## Location

- File: `src/rules/prefer-to-contain.ts`
- Function: default rule `create()` → `context.report({ fix ... })`
- Relevant code path:

```ts
fixer.removeRange([
  includesCall.callee.property.range[0] - 1,
  includesCall.range[1],
]),
```

## Problem

`isFixableIncludesCallExpression()` validates that the node is a single-argument, non-spread `includes()` call on a member expression — but it never checks `includesCall.callee.optional`. For optional calls:

```
arr?.includes('x')
      ^ property.range[0]
```

the character immediately before the `property` range is only the final `.` of the `?.` token; the `?` belongs to neither range. Removing `[property.range[0] - 1, includesCall.range[1])` deletes `.includes('x')` but keeps the `?`, producing `arr?`.

The subsequent replacements then build:

```js
expect(arr?).toContain('x');
```

which is a syntax error (optional token with no call/member following).

All four boolean/negation combinations (`toBe(true|false)` × `.not`) are affected whenever the receiver uses `?.`.

## Trigger / Reproduction

```js
// linted with prefer-to-contain: 'error'
expect(arr?.includes('x')).toBe(true);   // reported, --fix emits invalid code
expect(str?.includes('y')).toBe(false);  // same problem via .not.toContain path
```

## Expected Behavior

Either the fix should be suppressed for optional members (report without `fix`), or the removal should start at `includesCall.callee.object.range[1]` so the whole `?.includes(...)` portion is removed:

```js
fixer.removeRange([includesCall.callee.object.range[1], includesCall.range[1]])
```

This also naturally handles computed members if support is ever added.

## Actual Behavior

`--fix` rewrites valid source into a file containing `expect(arr?).toContain('x');`, which fails to parse.

## Impact

Users running `eslint --fix` (including editor save-fix and lint-staged setups) get a corrupted test file from an otherwise correct suggestion. Because nothing in the rule or its tests mentions optional chaining, every `arr?.includes(...)` expectation inside this rule's scope triggers the broken fix.

## Suggested Direction

Guard in `isFixableIncludesCallExpression`: return false when `node.callee.optional || node.optional` (and skip spread/computed as today), or switch the removal range to start at `callee.object.range[1]` so the entire accessor chain fragment is removed. A regression test with `expect(foo?.bar.includes('x')).toBe(true)` would pin it down.

## Evidence

- `src/rules/prefer-to-contain.ts`: the `removeRange([property.range[0] - 1, ...])` quoted above; no `callee.optional` check exists anywhere in the file.
- `src/rules/__tests__/prefer-to-contain.test.ts`: contains no optional-chaining cases (verified by search), so the gap is untested.

**Comment by G-Rath** [comment]
Please do not make reports based on static analysis alone - as it is, your analysis has failed to take into account the AST node structure: optional chains have a dedicated node (`ChainExpression`) that sits at the top of the tree for the argument so the rule will never report such code
